library/core/src/main/java/com/google/android/exoplayer2/audio/Ac4Util.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer2.audio;
+
+import com.google.android.exoplayer2.C;
+import com.google.android.exoplayer2.Format;
+import com.google.android.exoplayer2.drm.DrmInitData;
+import com.google.android.exoplayer2.util.MimeTypes;
+import com.google.android.exoplayer2.util.ParsableBitArray;
+import com.google.android.exoplayer2.util.ParsableByteArray;
+import java.nio.ByteBuffer;
+
+/**
+ * Utility methods for parsing AC-4 frames, which are access units in AC-4 bitstreams.
+ */
+public final class Ac4Util {
+
+  /**
+   * Holds sample format information as presented by a syncframe header.
+   */
+  public static final class SyncFrameInfo {
+
+    /**
+     * The sample mime type of the bitstream is {@link MimeTypes#AUDIO_AC4}.
+     */
+    public final String mimeType;
+    /**
+     * The bitstream version.
+     */
+    public final int bitstreamVersion;
+    /**
+     * The audio sampling rate in Hz.
+     */
+    public final int sampleRate;
+    /**
+     * The number of audio channels
+     */
+    public final int channelCount;
+    /**
+     * The size of the frame.
+     */
+    public final int frameSize;
+    /**
+     * Number of audio samples in the frame.
+     */
+    public final int sampleCount;
+
+    private SyncFrameInfo(
+        String mimeType,
+        int bitstreamVersion,
+        int channelCount,
+        int sampleRate,
+        int frameSize,
+        int sampleCount) {
+      this.mimeType = mimeType;
+      this.bitstreamVersion = bitstreamVersion;
+      this.channelCount = channelCount;
+      this.sampleRate = sampleRate;
+      this.frameSize = frameSize;
+      this.sampleCount = sampleCount;
+    }
+  }
+
+  /**
+   * The channel count of AC-4 stream.
+   */
+  // TODO: Parse AC-4 stream channel count.
```

Comment:
Is it feasible to do this as part of this change? It seems like it's important to expose an accurate channel count for surround audio.

Reply:
Thanks.
- Please could you make the constant private?
- If the input stream is not actually stereo (e.g., it's mono or 5.1) would you expect MediaCodec to throw an exception during configuration?

Reply:
Please could you make the constant private?
- sure.

If the input stream is not actually stereo (e.g., it's mono or 5.1) would you expect MediaCodec to throw an exception during configuration?
- No. The decoder itself is full functionality. That stream still can be decoded. In current implementation, decoder itself will ignore the input channel configuration information. (Decoder will parse this info by itself.)
Handling channel count is a little complex in AC-4, especially when we consider another concept in AC-4 -- AudioPresentation (https://developer.android.com/reference/android/media/AudioPresentation). This topic is under our internal discussion. I'll definitely submit a new pull request as long as it is done inside Dolby.